src/cosalib/meta.py

```diff
@@ -1,14 +1,91 @@
-from cosalib.build import _Build
+import json
+import os.path
 
+from cosalib.builds import Builds
+from cosalib.cmdlib import (
+    load_json,
+    write_json)
 
-class GenericBuildMeta(_Build):
+
+class GenericBuildMeta(dict):
     """
-    GenericBuildMeta  interacts with a builds meta.json
+    GenericBuildMeta interacts with a builds meta.json
     """
 
-    def __init__(self, build_dir, build, tmpbuilddir):
-        self.tmpbuilddir = tmpbuilddir
-        _Build.__init__(self, build_dir, build)
+    def __init__(self, build_dir, build='latest'):
+        builds = Builds(os.path.dirname(build_dir))
+        if build != "latest":
+            if not builds.has(build):
+                raise Exception('Build was not found in builds.json')
+        else:
+            build = builds.get_latest()
+
+        self._meta_path = os.path.join(
+            builds.get_build_dir(build), 'meta.json')
+        self.read()
+
+    def read(self):
+        """
+        Read the meta.json file into this object instance.
+        """
+        # Remove any current data
+        self.clear()
+        # Load the file
+        self.update(load_json(self._meta_path))
+
+    def write(self):
+        """
+        Write out the dict to the meta path.
+        """
+        write_json(self._meta_path, dict(self))
+
+    def get(self, key):
+        """
+        Returns the content of a dot key.
+
+        :param key: A key with dot delimiters
+        :type key: str
+        :returns: The value of the key
+        :rtype: any
+        :raises: KeyError
+        """
+        pathing = key.split('.')
+        loc = dict(self)
+        for p in pathing:
+            loc = loc[p]
+        return loc
+
+    def set(self, key, value):
+        """
+        Sets a dot delimited key to a value.
```

Comment:
Not a blocker, but I feel like "dot-delimited" should be (like in `jq`) syntax for a more general underlying join, since `.` is totally valid in JSON keys.

The obvious approach is to use varargs `*args` just like `os.path.join` e.g.; then this method would be just sugar that did `self.setv(value, key.split('.'))` or so.

Reply:
I like the idea. I'll give a shot. If I can put it together quickly I'll switch over!

Reply:
I have it working with a switch over. Cleaning up the code and updating tests.

Reply:
Updated.